Makefile.include

```diff
@@ -430,13 +430,13 @@ endif # RIOTNOLINK
 $(ELFFILE): $(BASELIBS)
 	$(Q)$(_LINK) -o $@
 
-$(BINDIR)/$(APPLICATION_MODULE).a: $(USEPKG:%=$(BINDIR)/%.a) $(BUILDDEPS)
+$(BINDIR)/$(APPLICATION_MODULE).a: $(BUILDDEPS) | pkg-build
 	$(Q)DIRS="$(DIRS)" "$(MAKE)" -C $(APPDIR) -f $(RIOTMAKE)/application.inc.mk
 $(BINDIR)/$(APPLICATION_MODULE).a: FORCE
 
 # Other modules are built by application.inc.mk and packages building
-_SUBMAKE_LIBS = $(filter-out $(BINDIR)/$(APPLICATION_MODULE).a $(USEPKG:%=$(BINDIR)/%.a) $(APPDEPS), $(BASELIBS))
-$(_SUBMAKE_LIBS): $(BINDIR)/$(APPLICATION_MODULE).a $(USEPKG:%=$(BINDIR)/%.a)
+_SUBMAKE_LIBS = $(filter-out $(BINDIR)/$(APPLICATION_MODULE).a $(APPDEPS), $(BASELIBS))
+$(_SUBMAKE_LIBS): $(BINDIR)/$(APPLICATION_MODULE).a | pkg-build
```

Comment:
@cladmi Does this correctly detect when a file is modified in the pkg?
This is an order-only prerequisite which is also a .PHONY, which is unfamiliar territory for me. We need to be certain that this does not cause make to assume that the target is up to date when the pkg files have been modified.

Reply:
@gebart For me `order-only` `.PHONY` prerequisite is something you execute everytime but you still check the file timestamp.
I used an 'order-only' because building it should not invalidate the other files timestamp, it is not its role here.

My train of thoughts here is:

* ELFFILE is rebuilt if files in `$(BASELIBS)` changed (but also depend on FORCE, so it is rebuilt everytime to ensure it is rebuilt if we change one of the variables)
* These lines say that "Files in `$(BASELIBS)` are generated by executing the `$(BINDIR)/$(APPLICATION_MODULE).a` and `pkg-build` targets.
* The target `$(BINDIR)/$(APPLICATION_MODULE).a`, that in fact generate most of the targets in `$(BASELIBS)`, currently depends on FORCE to rebuild it everytime
* The target `pkg-build` is `.PHONY` so rebuilt everytime

So now, they are all always executed.

I just put it `order-only` as for me, it is the same behavior as before, execute it every time (there was `$(USEPKG:%=$(BINDIR)/%.a): $(BUILDDEPS) FORCE` before) but still consider the generated files timestamps.

Currently it also does not really change anything as `$(BUILDDEPS)` contains `.PHONY` targets so forces rebuilding anyway. I may be able to change it to an order-only target, just a bit worried of the `make clean all -j` case.

Reply:
What I am confused about is how make treats the indirect dependencies via the order only prereq. Does make treat the .a file as needing a rebuild, or do we introduce a race here?
I know that the elf is always relinked in the current implementation of the build system, but that is not really what we want to do when nothing has changed, so it is necessary that the dependencies are correct or we are likely to introduce hidden problems that will bite us in the future.